libsrc/common/rand.s

```diff
@@ -54,18 +54,17 @@ _rand:  clc
         sta     rand+1
         adc     rand+2
         sta     rand+2
+        eor     rand+0
         and     #$7f            ; Suppress sign bit (make it positive)
         tax
         lda     rand+2
         adc     rand+3
         sta     rand+3
+        eor     rand+1
         rts                     ; return bit (16-22,24-31) in (X,A)
 
 _srand: sta     rand+0          ; Store the seed
         stx     rand+1
-        lda     #0
-        sta     rand+2          ; Set MSW to zero
-        sta     rand+3
-        rts
-
-
+        sta     rand+2          ; argument << 16 is convenient fill for MSW
+        stx     rand+3
+        jmp     _rand           ; to sufficiently "shuffle" first rand() result
```

Comment:
If _srand is placed before _rand then it can fall into _rand saving 3 bytes and cycles.

Reply:
Ah, good idea.